src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java

```diff
@@ -78,20 +78,17 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 
 	@Override
 	public boolean check(Event event) {
-		for (Object block : blocks.getArray(event)){
-			if (block instanceof Block) {
-				if (!items.check(event, (item) -> ((Block) block).isPreferredTool(item.getRandom()), isNegated()))
-					return false;
-			} else if (block instanceof BlockData) {
-				if (!items.check(event, (item) -> ((BlockData) block).isPreferredTool(item.getRandom()), isNegated()))
-					return false;
-			} else {
-				// invalid type
-				return false;
-			}
-		}
-		// all checks passed
-		return true;
+		return blocks.check(event,
+			block -> items.check(event,
+				item -> {
+					if (block instanceof Block) {
+						return ((Block) block).isPreferredTool(item.getRandom());
+					} else if (block instanceof BlockData) {
+						return ((BlockData) block).isPreferredTool(item.getRandom());
+					} else {
+						return false;
+					}
+				}), isNegated());
 	}
```

Comment:
Say I changed `return false;` to `return isNegated();` that should fail for if both are null yes?

Reply:
The latter is expected to return `true` due to negation

Reply:
yeah didn't matter anyways when I tried test couldn't even run, for now just gonna fix the fact block data test don't actually use block data